tests/release-cycle-readiness-head/clone_repositories.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# Clones the repositories we check in the workspace directory
+# (/tmp/release-readiness). It's convenient to separate the data
+# retrieval part (this script) and the logic (check_readiness.sh)
+# separated, so that the maitnainers can iterate the logic without
+# needing cloning the repositories.
+
+set -ef
+
+WORK_DIR=/tmp/release-readiness
+mkdir -p ${WORK_DIR}
+
+function clone_repository() {
+  # e.g., java-bigtable
+  repo_name=$1
+  revision=$2
+
+  repo_dir="${WORK_DIR}/${repo_name}"
+  if [ ! -d "${repo_dir}" ]; then
+    repo_url="https://github.com/googleapis/${repo_name}"
+    # Clone the revision
+    if [ -z "${revision}" ]; then
```

Comment:
Is the `revision` really used in the shell?

Reply:
It's used in `git checkout ${revision}`

Reply:
I think the `revision` variable is always empty during the function call.

Reply:
The logic was unnecessarily complex. I removed one if-statement.